Add draft release notes for 2.10.17 (#10036)
PrefectHQ/prefect: RELEASE-NOTES.md

# Prefect Release Notes

> ## Release 2.10.17

Reviewer: Making we can take this opportunity to celebrate our new contributors since we have more than usual?

Otherwise we could highlight our new and improved tutorial!

Author: I added a highlight for the tutorial.

### Enhancements

> - Update PrefectClient to follow redirects by default — https://github.com/PrefectHQ/prefect/pull/9988

Reviewer: ```suggestion
- Update Prefect client to follow redirects by default — https://github.com/PrefectHQ/prefect/pull/9988
```

Reviewer: or `PrefectClient` but not used as often in our notes

> - Update PrefectClient to follow redirects by default — https://github.com/PrefectHQ/prefect/pull/9988

Reviewer 2: ```suggestion
- Update PrefectClient to follow redirects by default — https://github.com/PrefectHQ/prefect/pull/9988
- Always show checkboxes rather than animating them on hover — https://github.com/PrefectHQ/prefect-ui-library/pull/1490
```

> - Add the CustomWebhookNotificationBlock for general purpose notification — https://github.com/PrefectHQ/prefect/pull/9547

Reviewer: ```suggestion
- Add the `CustomWebhookNotificationBlock` for general purpose notification — https://github.com/PrefectHQ/prefect/pull/9547
```

> - Add the CustomWebhookNotificationBlock for general purpose notification — https://github.com/PrefectHQ/prefect/pull/9547

Reviewer 3: ```suggestion
- New `CustomWebhookNotificationBlock` for triggering custom webhooks in response to flow run state changes — https://github.com/PrefectHQ/prefect/pull/9547
```

### Fixes

> - Resolve bug that could throw an OSError if too many files were being scanned for flows — https://github.com/PrefectHQ/prefect/pull/10014

Reviewer 3: ```suggestion
- Limit the number of files concurrently opened by `prefect deploy` when searching for flows — https://github.com/PrefectHQ/prefect/pull/10014
```

> - Resolve bug that could throw an OSError if too many files were being scanned for flows — https://github.com/PrefectHQ/prefect/pull/10014

Reviewer: ```suggestion
- Resolve bug that could throw `OSError` if too many files were being scanned for flows — https://github.com/PrefectHQ/prefect/pull/10014
```

> - Limit the number of files concurrently opened by `prefect deploy` when searching for flows — https://github.com/PrefectHQ/prefect/pull/10014

Reviewer 2: ```suggestion
- Limit the number of files concurrently opened by `prefect deploy` when searching for flows — https://github.com/PrefectHQ/prefect/pull/10014
- Fix js error when running prefect in an unsecure context (http vs https) — https://github.com/PrefectHQ/prefect-ui-library/pull/1501
```

Reviewer: @pleek91 Can you rephrase 
"Fix js error when running prefect in an unsecure context (http vs https)" as what that means for the user?

Reviewer: What page does it cause to break in the Prefect UI?

Reviewer 2: Just the ui in general, its not specific to a single page

### Documentation

> - Fix broken link in deployments UX page — https://github.com/PrefectHQ/prefect/pull/10013

Reviewer 3: ```suggestion
- Fix broken link to `prefect-docker` documentation on the deployments UX page — https://github.com/PrefectHQ/prefect/pull/10013
```

> - Document --work-queue / -q arguments to `worker start` command — https://github.com/PrefectHQ/prefect/pull/10027

Reviewer: ```suggestion
- Document `--work-queue / -q` arguments to `worker start` command — https://github.com/PrefectHQ/prefect/pull/10027
```

> - Add a link to join Club 42 to the Community page — https://github.com/PrefectHQ/prefect/pull/9927

Reviewer: ```suggestion
- Add link to join Club 42 to Community page — https://github.com/PrefectHQ/prefect/pull/9927
```

> - Update the tutorial to be more succinct and purposeful  — https://github.com/PrefectHQ/prefect/pull/9940

Reviewer: ```suggestion
- Update Prefect tutorial to be more succinct and purposeful  — https://github.com/PrefectHQ/prefect/pull/9940
```

> - Update the tutorial to be more succinct and purposeful  — https://github.com/PrefectHQ/prefect/pull/9940

Reviewer 3: This PR seems like a good one to use for the highlight section.

# Prefect Release Notes

## Release 2.10.17

### Improved Prefect Tutorial

> Unleash the full potential of Prefect with the [newly improved tutorial](https://docs.prefect.io/2.10.17/tutorial/). Elevate your learning experience and embark on a transformative journey that equips you with the skills to orchestrate and observe all of your workflows.

Reviewer 4: ```suggestion
Prefect's documentation has a [new and improved tutorial](https://docs.prefect.io/2.10.17/tutorial/). The tutorial focuses on the key elements and features of Prefect. You can find content related to more advanced features or less common use cases in the [Guides](https://docs.prefect.io/2.10.17/guides/) section
```

> Unleash the full potential of Prefect with the [newly improved tutorial](https://docs.prefect.io/2.10.17/tutorial/). Elevate your learning experience and embark on a transformative journey that equips you with the skills to orchestrate and observe all of your workflows.

Reviewer: ```suggestion
We're excited to announce an enhanced tutorial, designed to showcase the latest features in Prefect. With the introduction of concepts like work pools and workers, alongside an interactive deployment CLI, this tutorial reflects the elevated experience that these new features bring.

Follow along as we simplify the entire process, providing a seamless experience from creating a flow to having a worker execute your flow. By the end of this tutorial, you'll have gained a comprehensive understanding of Prefect's core concepts and be equipped to harness their power effectively. You'll learn to:

1. Createa Flow: Learn how to define and structure your workflows using Python code, with Prefect's intuitive syntax.
2. Add Tasks: Discover how to incorporate tasks into your flow, enabling the execution of specific actions or computations.
3. Create a Work Pool: Explore the concept of work pools, which allow you to distribute tasks across multiple workers efficiently.
4. Deploy a Worker: Gain hands-on experience in deploying a worker, a vital component that executes tasks within your flow.
5. Deploy the Flow: Learn how to deploy your flow, ensuring it is readily available for execution on your worker.
6. Run the Flow on the Worker: Run your flow seamlessly on the worker.
```

### Enhancements
- Update Prefect client to follow redirects by default — https://github.com/PrefectHQ/prefect/pull/9988

> - Always show checkboxes rather than animating them on hover — https://github.com/PrefectHQ/prefect-ui-library/pull/1490

Reviewer 4: ```suggestion
- Always show checkboxes on list items, rather than animating them on hover — https://github.com/PrefectHQ/prefect-ui-library/pull/1490
```
